vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/experiments/ExperimentListComponent.java

```diff
@@ -0,0 +1,193 @@
+   * <p>
+   * This method should be used to provide the experiments within a
+   * {@link life.qbic.projectmanagement.domain.project.Project} to {@link ExperimentListComponent}
+   */
+  public void setExperiments(Collection<Experiment> experiments) {
+    experimentItemCollection.removeAll();
+    addItemCollectionListeners();
+    experiments.forEach(experiment -> experimentItemCollection.addExperimentItem(
+        ExperimentItem.create(experiment)));
+  }
+
+  /**
+   * Provides the {@link ExperimentId} which annotates the currently active Experiment to this
+   * component
+   * <p>
+   * This informs the {@link ExperimentItemCollection} about which experiment is set as active via
+   * the provided {@link ExperimentId}
+   */
+  public void setActiveExperiment(ExperimentId experimentId) {
+    experimentItemCollection.findBy(experimentId).ifPresent(ExperimentItem::setAsActive);
+  }
+
+  /**
+   * Provides the {@link ExperimentId} which annotates the currently selected Experiment to this
+   * component
+   * <p>
+   * This informs the {@link ExperimentItemCollection} about which experiment was selected by the
+   * user via the provided {@link ExperimentId}
+   */
+  public void setSelectedExperiment(ExperimentId experimentId) {
+    experimentItemCollection.findBy(experimentId).ifPresent(ExperimentItem::setAsSelected);
+  }
+
+  /**
+   * Adds the provided {@link ExperimentSelectionListener} to the list of listeners which will
+   * retrieve notification if an {@link Experiment} was selected within this component
+   */
+  public void addExperimentSelectionListener(
+      ExperimentSelectionListener experimentSelectionListener) {
+    this.selectionListeners.add(experimentSelectionListener);
+  }
+
+  /**
+   * Adds the provided {@link ExperimentCreationListener} to the list of listeners which will
+   * retrieve notification if a new {@link Experiment} was created in this component
+   */
+  public void addExperimentCreationListener(ExperimentCreationListener experimentCreationListener) {
+    this.creationListener.add(experimentCreationListener);
+  }
+
+  private void fireExperimentCreatedEvent(ExperimentCreationEvent event) {
+    creationListener.forEach(it -> it.handle(event));
+  }
+
+  private void fireExperimentalItemSelectedEvent(ExperimentItemClickedEvent event) {
+    selectionListeners.forEach(it -> it.handle(event));
+  }
+
+  private void displaySuccessfulExperimentCreation() {
+    SuccessMessage successMessage = new SuccessMessage("Experiment Creation succeeded", "");
+    StyledNotification notification = new StyledNotification(successMessage);
+    notification.open();
+  }
+
+  private void displayExperimentCreationFailure() {
+    ErrorMessage errorMessage = new ErrorMessage("Experiment Creation failed", "");
+    StyledNotification notification = new StyledNotification(errorMessage);
+    notification.open();
+  }
+
+  @FunctionalInterface
+  public interface ExperimentSelectionListener {
+
```

Comment:
for interface methods, JavaDoc is essential

Reply:
Good point it's done 🤔